pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -131,6 +135,24 @@ func (r *pullthroughBlobStore) ServeBlob(ctx context.Context, w http.ResponseWri
 	return nil
 }
 
+// Create needs to be overridden in order to use pullthroughBlobStore middleware during cross-repository push so that remote layers don't need to be pushed.
+func (r *pullthroughBlobStore) Create(ctx context.Context, options ...distribution.BlobCreateOption) (distribution.BlobWriter, error) {
+	options = append(options, storage.WithRepositoryMiddlewareWrapper(
+		func(ctx context.Context, repo distribution.Repository, name reference.Named) (distribution.Repository, error) {
+			nameParts := strings.SplitN(name.Name(), "/", 2)
```

Comment:
do we want to artificially limit the segments to 2, or fail if they give us something with more than 2 segments?